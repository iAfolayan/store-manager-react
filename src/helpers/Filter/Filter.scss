.filter {
  margin: 10px 0;
  height: 60px;
  border-bottom: 1px solid #ccc;
  padding: 6px;
  margin-right: 1rem;
  clear:  both;
  max-width: 97%;
}
.filter input {
  float: right;
}