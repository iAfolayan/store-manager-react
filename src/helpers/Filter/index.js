import React, { Component } from 'react';
import './Filter.scss';
import { connect } from 'react-redux';
// import Cart from './image/cart.png';

export class Filter extends Component {
	renderMenu() {
		return (
			<div id="cartHolder">
				<img src='./image/cart.png' alt="" />
			</div>
		);
	}

	render() {
		const { handleChange, value} = this.props;
		const { response } = this.props.auth;
		return (
			<div className="col-sm-12 filter">
				<form className="form-group">
					<input
						type="search"
						className="form-control col-md-3"
						placeholder="Filter product by name"
						onChange={handleChange}
						value={value}
					/>
					{ response.role === 1 ? '' : this.renderMenu() }
				</form>
				
			</div>
		);
	}
}

const mapStateToProps = state => ({
  auth: state.authReducer
});

export default connect(
  mapStateToProps,
)(Filter);