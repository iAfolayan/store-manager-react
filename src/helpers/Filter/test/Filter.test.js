import React from 'react';
import { shallow } from 'enzyme';
import Filter from '..';

describe('Filter Component', () => {
  test('should render filter component without crashed', () => {
    const wrapper = shallow(<Filter />);
    expect(wrapper).toMatchSnapshot();
  })

  test('should test onchange event', () => {
    const wrapper = shallow(<Filter />);
    wrapper.instance().handleChange();
    expect(wrapper).toBeDefined();
  })

 test('should simulate filter', () => {
    const wrapper = shallow(<Filter />);
    wrapper.instance().filterproduct();
    expect(wrapper).toBeDefined();
  });
});
