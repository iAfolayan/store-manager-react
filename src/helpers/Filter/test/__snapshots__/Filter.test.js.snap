// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component should render filter component without crashed 1`] = `
<div
  className="col-sm-12 filter"
>
  <form
    className="form-group"
  >
    <input
      className="form-control col-md-3"
      placeholder="Filter product by name"
      type="search"
    />
  </form>
</div>
`;
